### ISSUE TITLE
No documentation for API users. 

### ISSUE THREAD

**Issue by wammar** [higgood/seechat-folks#49]
**Context**
The Prosper project critically relies on the SeeChat x Ideas APIs.

**Describe the bug**
There's no clear documentation for authenticating users through the API endpoints.

**[Required] Where did you encounter the bug**
None

**[Required] Reality vs. Expectation**
Reality: https://api.staging.seechat.ai/docs does not provide endpoints for authenticating users.
Expectation: platforms which depend on the API should be able to authenticate users through the API.

**Screenshots**
N/A

Thank you for improving the Idea Factory!!

**Comment by wammar** [comment]
q: is this the expected behavior?
@Mox301 addressed this in a recent PR and currently being reviewed by @muhammadnasr

**Comment by wammar** [comment]
@muhammadnasr there's an existing task for this at https://github.com/higgood/seechat/issues/207

**Comment by wammar** [comment]
Duplicate of https://github.com/higgood/seechat/issues/207

**Comment by muhammadnasr** [comment]
How much credit should the API user have, 3 like other users or what?

**Comment by wammar** [comment]
Yes, please use the same credit system. In the future, we will have group-level control but this is not necessary at the moment. Any user, regardless of which interface they're using for the SeeChat x Ideas backend (ie, the API or the seechat.ai/ideas) should have the same treatment.

**Comment by muhammadnasr** [comment]
Duplicate of https://github.com/higgood/seechat/issues/207

**Comment by wammar** [comment]
Lowering the priority of this task to focus on the competition use case.

**Comment by Ahmed-Habashy** [comment]
The issue had been discussed many times (https://github.com/higgood/seechat/issues/207, https://github.com/higgood/seechat/issues/416) and it is aligned with 2025 Q1/O3 KR3

**Comment by wammar** [comment]
@Mox301 Please investigate in 1 hour.

**Comment by nova-sudo** [comment]
Recording:
https://drive.google.com/file/d/1uJtOH4tUIt0pzp61DeNIMOXYYS-azLFY/view?usp=drive_web
https://drive.google.com/file/d/1urwjHy5kk60OpI_7wEhjop2Oa02yeuKV/view?usp=drive_web

Notes
https://docs.google.com/document/d/1Dd5xEyXdp4JKjR_OkrN5uWjUOFJgeI-BuQkYmNxUTZ4/edit?usp=drive_web

**Comment by Ahmed-Habashy** [comment]
We’ve attached the initial API documentation for the /idea/create_hypothesis endpoint, which allows external users to submit idea-related data using an API key. Mostafa has successfully tested this API and confirmed it works as expected for sending the problem and scientific area from external sources.

However, based on required bot usage scenarios, we’ve identified the need to extend the API to optionally support additional metadata — particularly the abstract, authors, and source link — when the submission is coming from a bot.

We’ll revise and expand the current API documentation accordingly, and any changes will also be reflected in the OpenAPI schema at /docs.

Documentation (v1) is attached for review.

[External-createhypothesis-access-instructions.pdf](https://github.com/user-attachments/files/19974091/External-createhypothesis-access-instructions.pdf)

**Comment by nova-sudo** [comment]
Working on this issue, the solution is clear, but it will block Mostafa's work on the automated idea import to Seechat x Ideas.

**Comment by wammar** [comment]
@nova-sudo: We need to change the pydantic model for the endpoint /user-info and modify the medrxiv bot on seechat-folks accordingly.

**Comment by Ahmed-Habashy** [comment]
@nova-sudo 
### [⚠️ Update] Add Limitation About Sequential Hypothesis API Usage

While testing the `/idea/create_hypothesis` and `/idea/edit_hypothesis` endpoints using the medRxiv bot, we noticed an important limitation: If `edit_hypothesis` is called immediately after `create_hypothesis`, it may fail because the backend hasn't finalized the creation process yet.

💡 Please add a note to the API documentation like:
> "After calling `create_hypothesis`, allow a short delay before invoking `edit_hypothesis`. The backend needs time to complete object instantiation before edits are accepted."

This will help prevent bot or script-based clients from encountering confusing failures when editing hypotheses.
